src/utilities/graphql/fragments.ts

```diff
@@ -143,3 +144,26 @@ export function getFragmentFromSelection(
       return null;
   }
 }
+
+export function isFullyUnmaskedOperation(document: DocumentNode) {
+  let isUnmasked = true;
+
+  visit(document, {
+    FragmentSpread: (node) => {
+      if (!node.directives) {
+        isUnmasked = false;
+        return BREAK;
+      }
+
+      isUnmasked &&= node.directives.some(
```

Comment:
```suggestion
      isUnmasked = node.directives.some(
```

I think we can safely assume that at this point, `isUnmasked` is always `true`, right?